app/routes/chatbot.js

```diff
@@ -44,11 +41,41 @@ function isCommand(incomingMessage, commandType) {
 }
 
 /**
+ * Renders message for given messageType, sends it to continue conversation for current campaign.
+ * Assumes we have a loaded req.user and req.campaign.
+ */
+function continueConversationWithMessageType(req, res, messageType) {
+  const scope = req;
+
+  return contentful.renderMessageForPhoenixCampaign(req.campaign, messageType)
+    .then((message) => {
+      scope.replyMessage = helpers.addSenderPrefix(message);
+      // Store current campaign for subsequent messages.
+      scope.user.current_campaign = req.campaignId;
+
+      return scope.user.save();
+    })
+    .then(() => {
+      logger.debug(`saved user:${req.user._id} current_campaign:${req.campaignId}`);
+
+      // todo: Promisify this POST request and only send back Gambit 200 on profile_update success.
+      const oip = process.env.MOBILECOMMONS_OIP_CHATBOT;
+      req.user.postMobileCommonsProfileUpdate(oip, scope.replyMessage);
+      stathat.postStat(`campaignbot:${messageType}`);
+      BotRequest.log(req, 'campaignbot', messageType, scope.replyMessage);
+
+      return helpers.sendResponse(res, 200, scope.replyMessage);
+    })
+    .catch(err => helpers.sendErrorResponse(res, err));
+}
+
+/**
  * Sends given message to user to end conversation.
  */
 function endConversationWithMessage(req, res, message) {
-  // TODO: Promisify send_message and return 200 when we know message delivery successful.
-  mobilecommons.send_message(req.user.mobile, message);
+  // todo: Promisify this POST request and only send back Gambit 200 on profile_update success.
+  const oip = process.env.MOBILECOMMONS_OIP_AGENTVIEW;
+  req.user.postMobileCommonsProfileUpdate(oip, message);
```

Comment:
Confirmed that I replied to a Closed Campaign Message to confirm my response was sent to Agent View, and I show in Agent View up as the CampaignBot: Production. Although I'm thinking maybe we can use `send_message`, and just post the Mobile Commons Campaign ID for the CampaignBot MC Campaign instead of the Mobile Commons MC Campaign. We'd get a small performance win there, although we'd lose the ability to see the last Gambit Chatbot Response sent on the profile field.


<img width="1013" alt="screen shot 2017-04-14 at 9 38 55 am" src="https://cloud.githubusercontent.com/assets/1236811/25049503/ff335b12-20f6-11e7-9635-2f26c3f4563f.png">

@sergii-tkachenko @deadlybutter @jamjensen -- Was there a reason behind choosing the Mobile Commons MC Campaign instead of the CampaignBot MC Campaign when the `send_message` function was implemented in #763?